zipline/gens/brokers/robinhood_broker.py

```diff
@@ -0,0 +1,546 @@
+                if not last_price:
+                    # Lets try again
+                    last_price = iex.get_lastSalePrice(symbol)
+
+                created = utc_to_local(datetime.strptime(result["created_at"], "%Y-%m-%dT%H:%M:%S.%fZ"))
+                cost_basis = float(result["average_buy_price"])
+                
+                positions[symbol_lookup(symbol)] = Position(amount=amount,
+                                                      cost_basis=cost_basis,
+                                                      last_price=last_price,
+                                                      created=created)
+
+                # position_value = position_value+(cost_basis*amount)
+                if amount > 0:
+                    unrealized_pl = unrealized_pl + ((last_price * amount) - (cost_basis * amount))
+                    long_position_value = long_position_value + (cost_basis * amount)
+                else:
+                    unrealized_pl = unrealized_pl + ((cost_basis * np.abs([amount])[0]) - (last_price * np.abs([amount])[0]))
+                    short_position_value = long_position_value + (cost_basis * np.abs([amount])[0])
+
+        pnl = equity-uncleared_deposits-yesterday_equity  # unrealized_pl + unsettled_funds
+        leverage = 0
+        net_leverage = 0
+        if portfolio_value > 0:
+            leverage = (long_position_value + short_position_value) / portfolio_value
+            net_leverage = market_value / portfolio_value
+
+        portfolio = Portfolio()
+        portfolio.capital_used = np.abs([(short_position_value - long_position_value)])[0]
+        portfolio.cash = total_cash
+        portfolio.pnl = pnl
+        portfolio.positions = positions
+        portfolio.portfolio_value = portfolio_value
+        portfolio.positions_value = market_value
+        portfolio.returns = returns
+        portfolio.starting_cash = self._starting_cash
+        portfolio.start_date = self._start_date
+
+        self.portfolio = portfolio
+
+        account = Account()
+        # account.accrued_interest=acct_info
+        account.available_funds = portfolio_value
+        account.buying_power = buying_power
+        account.cushion = total_cash / portfolio_value
+        account.day_trades_remaining = float("inf")
+        account.equity_with_loan = portfolio_value
+        account.excess_liquidity = port_info["excess_margin"]
+        account.initial_margin_requirement = float(acct_info["margin_balances"]["margin_limit"]) if "margin_balances" in acct_info and "margin_limit" in acct_info["margin_balances"] else 0
+        account.leverage = leverage
+        account.maintenance_margin_requirement = portfolio_value-float(port_info["excess_margin"])
+        account.net_leverage = net_leverage
+        account.net_liquidation = portfolio_value
+        account.settled_cash = cash
+        account.total_positions_value = market_value
+
+        self.account = account
+
+
+class ROBINHOODBroker(Broker):
```

Comment:
Most of this code is duplicate of IBBroker. We should consider moving up the shared code to Broker to avoid code duplication. Could you please check what's possible?